src/helpers/3box.ts

```diff
@@ -1,14 +1,63 @@
 import { subgraphRequest } from '@snapshot-labs/snapshot.js/src/utils';
 
 export async function getProfiles(addresses) {
-  return subgraphRequest('https://api.3box.io/graph', {
-    profiles: {
-      __args: {
-        ids: addresses
-      },
-      name: true,
-      eth_address: true,
-      image: true
+  const [ensProfiles, threeBoxProfiles] = await Promise.all([
+    subgraphRequest('https://api.thegraph.com/subgraphs/name/ensdomains/ens', {
+      accounts: {
+        __args: {
+          first: 1000,
+          where: {
+            id_in: addresses.map(address => address.toLowerCase())
+          }
+        },
+        id: true,
+        domains: {
+          __args: {
+            first: 1
+          },
+          name: true
+        }
+      }
+    }),
+    subgraphRequest('https://api.3box.io/graph', {
+      profiles: {
+        __args: {
+          ids: addresses
+        },
+        name: true,
+        eth_address: true,
+        image: true
+      }
+    })
+  ]);
+
+  const profiles = {};

```

Comment:
This looks like a lot of code for doing a small thing, if you do the changes below you could simplify here and do something like this:
```
return Object.from.entries(Object.entries(profiles).map((address, profile) => {
  profile = _3BoxProfiles[address] || {};
  profile.ens = ensNames[address];
  return [address, profile];
}));
```